pkg/awsiamauth/hook.go

```diff
@@ -0,0 +1,51 @@
+package awsiamauth
+
+import (
+	"context"
+	"errors"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/workflow"
+	"github.com/aws/eks-anywhere/pkg/workflow/management"
+	"github.com/aws/eks-anywhere/pkg/workflow/workflowcontext"
+)
+
+type HookRegistrar struct {
+	*AwsIamAuth
+	Spec *cluster.Spec
```

Comment:
For structs with tons of members I can buy that it makes the code a bit cleaner, but for this I would vote to have a constructor and make the members private.

It makes clear what is mandatory and what's not. Also, it stops anyone from tinkering with the inner members directly once the struct is constructed.

Reply:
For `Spec` in particular I'm not sure why I did it this way - will adjust. 

The `AwsIamAuth` is embedded so will have to remain exposed. That's done intentionally for convenience so consumers can use the registrar as anything satisfying existing AWS IAM interfaces.

Reply:
Done